src/message.cpp

```diff
@@ -1318,9 +1328,28 @@ string message::format_many_ids(const string& id)
 
 vector<string> message::parse_many_ids(const string& ids)
 {
-    vector<string> idv;
+	vector<string> idv;
+	const regex r(R"(<([a-zA-Z0-9\!#\$%&'\*\+\-\./=\?\^\_`\{\|\}\~]+)\@([a-zA-Z0-9\!#\$%&'\*\+\-\./=\?\^\_`\{\|\}\~]+)>)");
+	if (!_strict_mode && ids.empty())
+	{
+		return {};
+	}
+	else if (!_strict_mode)
+	{
+		auto start = ids.cbegin();
+		auto end = ids.cend();
+		match_flag_type flags = match_default;
+		match_results<string::const_iterator> what;
+		while (regex_search(start, end, what, r, flags))
+		{
+			idv.push_back(what[0]);
+			start = what[0].second;
+			flags |= match_prev_avail;
+			flags |= match_not_bob;
```

Comment:
Is it necessary to set the flags in each loop iteration? In addition, are these flags necessary at all or the default one plus the not null flag is good enough?